ecmascript/parser/src/parser/stmt/module_item.rs

```diff
@@ -60,8 +60,13 @@ impl<'a, I: Tokens> Parser<'a, I> {
             .map(ModuleItem::from);
         }
 
-        let type_only_span = self.input.cur_span();
-        let type_only = self.syntax().typescript() && eat!("type");
+        let type_only = self.input.syntax().typescript()
+            && is!("type")
+            && (peeked_is!('{') || !peeked_is!("from") && !peeked_is!(','));
```

Comment:
Wait... `import type from from "./a"` 🤯

That's actually a bug in the typescript compiler.

Reply:
I'm not sure about a way to do a peek twice or what would be best to fix this. Overall it's not too important and such an edge case... going to sleep now.

Reply:
I think the best solution is just ignoring it.
I don't expect a user to write code like `import type from 'foo';`